### PR TITLE
Adjust overlay styles

### PR DESCRIPTION
**Pull request by smbea**
Related to #2488

- Change border and box-shadow
- Change overlay title style
- Adjust report feedback margin around line divider

Artifact: https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/update-overlay-styles/camunda-modeler-update-overlay-styles-mac.dmg

⚠️ waiting to be integrated with #2572

### CODE REVIEW

**Comment by nikku** [comment]
Please update this on top of https://github.com/camunda/camunda-modeler/pull/2571.

If we fix anything margin related, we shall do that in the generic, newly introduced section styles, not locally in a component.

**Comment by andreasgeier** [comment]
I believe it makes sense for me to wait with the design review until you implemented the changes requested by Nico.

However, these issues I can already report:

- Ensure that all links in overlays are underlined as it is already the case in the cloud connect overlay. Including the two options for start instance.
- Bold text in the What's new message is using font-weight:700 (native user agent style). Please ensure font-weight:500 as we have for the titles in the overlays.
- Remove the arrow from the learn more link on the profile overlay. This is outdated, we don't use this icon anymore for external links in order to be consistent with other places like properties panel and to avoid misalignment with inline links in paragraphs. Also, remove the extra section for it and put the link to the description.*

_*I try to deliver a design example asap_

**Comment by smbea** [comment]
Will wait on #2572 to merge this one so we can test the final look with all the overlays.

File: client/src/plugins/version-info/ReleaseInfo.less [hunk 1]
```diff
@@ -1,9 +1,10 @@
 :local(.ReleaseInfo) {
   overflow: auto;
+  max-height: 500px;
```
Comment (nikku) [inline client/src/plugins/version-info/ReleaseInfo.less:3]: Release info max height shall be limited by the parent container (cf. Section#maxHeight).

File: client/src/plugins/version-info/ReleaseInfo.less [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
 
   h1, h2, h3, h4 {
     font-size: inherit;
-    font-weight: bold;
+    font-weight: 500;
```
Comment (nikku) [inline client/src/plugins/version-info/ReleaseInfo.less:7]: Is that the default font weight? Let us not deal with magic numbers here but instead (re)set the font weight in a single place.

File: client/src/plugins/version-info/ReleaseInfo.less [hunk 3]
```diff
@@ -17,4 +18,20 @@
       margin-bottom: 0;
     }
   }
+
+  div {
```
Comment (nikku) [inline client/src/plugins/version-info/ReleaseInfo.less:22]: Why a `div` here? What shall we accomplish? 

My proposal is to make this an actual heading instead, i.e. `h4` to match semantic HTML.

File: client/src/plugins/version-info/ReleaseInfo.less [hunk 4]
```diff
@@ -17,4 +18,20 @@
...
+    font-weight: 500;
+    margin-bottom: 6px;
+  }
+
+  ul {
```
Comment (nikku) [inline client/src/plugins/version-info/ReleaseInfo.less:27]: If this is the new way to style lists, let us define it globally, in a shared space and re-use it across all components.

**Review by nikku — CHANGES_REQUESTED** [review]
Feedback on the what's new communcation UI; also applies to any kinds of lists I suppose? My appologies for sending that in so late!

* List items are hard to distinguish (each item shall be grouped) => List items shall be grouped visually (I'd assume: different offset between header and content + next list item)
* Font size / style difference between headers and text is not distinguishable on lower resolution devices / with less sight => I'd aim for clarity instead of being fancy here
* Why do we use `-` for lists? I do not see that being used anywhere on the web or apps I use. (I find it odd, personally and would rather stay clear and standard here (*))

(Just sharing my feedback here. Generally speaking I'd aim to be "not fancy" for all kinds of basic text formatting to not get into the way of our users; CC @andreasgeier). 

---

Feedback on the implementation:

* Ensure that styles are defined in a single place and make them re-usable; no magic numbers, use variables instead.
* Perspective to assume: Whatever we implement, implement it in a way that we can adjust in a single place, quickly, in the future.

---

Screenshot of the current design on Linux:

image

**Comment by nikku** [comment]
Cf. also https://github.com/camunda/camunda-modeler/issues/2488#issuecomment-976831728; we definitely shall extract all basic styles into a single place and re-use.

**Comment by smbea** [comment]
I extracted the ul and ol (used as suggestion in the example example) styles and the font-weight for the headings (the new bold weight) to the `overlay.less` file.

**Comment by andreasgeier** [comment]
> Feedback on the what's new communcation UI

@nikku I understand some of your concerns. But keep in mind that this `What's new` message in the overlay is the first increment of the feature. We can improve the content and the style but I'd prefer to tackle this as a separate issue. For the next increment, I can imagine having this message in a virtual tab instead of the limited overlay. Having said this, I'll try out if we can improve the readability of the bullet points with a bit of spacing.

File: client/src/plugins/version-info/ReleaseInfo.less (hunk 4 above)
Comment (andreasgeier) [reply client/src/plugins/version-info/ReleaseInfo.less]: I think the `What's new` message is not a globally used style. For such a (long) message, we might have more whitespace and other alignments as in other places, for instance, the file list or a list in a property description.

File: client/src/shared/ui/Section.less [hunk 5]
```diff
@@ -6,20 +6,20 @@
   max-height: var(--section-max-height);
   flex: inherit;
 
-  .section__header,
   .section__body {
-    padding: 12px;
+    padding: 12px 12px 0 12px;
+    margin-bottom: 12px;
```
Comment (nikku) [inline client/src/shared/ui/Section.less:11]: Why? What do we want to accomplish with this one?

File: client/src/plugins/version-info/ReleaseInfo.less (hunk 4 above)
Comment (nikku) [reply client/src/plugins/version-info/ReleaseInfo.less]: My two cents is: Lists shall be designed in the same way everywhere. At least this shall be our default approach (start from being generic and specialize from there).

What's your opinion on this @andreasgeier?

File: client/src/shared/ui/Section.less (hunk 5 above)
Comment (smbea) [reply client/src/shared/ui/Section.less:11]: The padding is applied at the end of that div. Which means when it's not fully scrolled, there isn't any whitespace. See left (padding) and right (margin):
<div>



 </div>

Comment (nikku) [reply client/src/shared/ui/Section.less:11]: My expectation as a user is to not have that padding, but have the scroll bar to the end of the section container. 

It complicates our styles and it does not match the default UX is see in apps and browsers around. Please clarify with @andreasgeier and re-consider potentially.

Comment (nikku) [reply client/src/shared/ui/Section.less:11]: Section at the moment is built in a way that it works as expected in browsers and apps (no bottom padding).

Comment (andreasgeier) [reply client/src/shared/ui/Section.less:11]: The basic rule here is that the layout should not change on overflow. When we have whitespace between a section start/end and its content, we keep that space in the overflow scenario. Another example is the file list for deployment. So, let's ensure we have this expected behavior in place.

https://user-images.githubusercontent.com/938244/143227596-48452007-c2df-4443-a908-7095e591dbd4.mov

Comment (nikku) [reply client/src/shared/ui/Section.less:11]: What about the following screen: 

capture ehjNzh_optimized

You clearly see that we do have the expected behavior (from my point of view) in place in  place everywhere.

This is how I'd imagine it to work for sections, too: 

capture IQsYnK_optimized


Note that it also does not break the additional files list you're mentioning:

capture 3Zt4Wo_optimized

On MacOS this is less visible; on any other platform having a padding below a scrollable container is visual disturbing.

Comment (andreasgeier) [reply client/src/shared/ui/Section.less:11]: @nikku Your examples are different scenarios. You can't compare a table scrolling with a content section. I believe the file list example makes it very clear, I don't want to have this list docked on the deploy button. 
I hear and appreciate your feedback. But, let's conclude this discussion and keep it as designed.

Comment (nikku) [reply client/src/shared/ui/Section.less:11]: Thanks for clarifying :+1:

**Review by nikku — APPROVED** [review]
O.K. from my side for the moment.

Let us assess the current state next week and investigate further technical refinements once everything is in place.

**Review by andreasgeier — APPROVED** [review]
Thanks @smbea for the remote session to fix the last remaining styling issues. 

✅ Approved

**Pull request merged**